07-contributor-guide.Rmd

```diff
@@ -38,515 +46,253 @@ Following good software development and coding practices simplifies collaboratio
 
-* To describe the kind of work to be done: bug, enhancement, task, discussion, question, suitable for beginners
-* To indicate the state of the issue: urgent, current, next, eventually, won't fix, duplicate
+```{r, echo=FALSE}
+DiagrammeR::mermaid("
+        graph TB
+        A[Author: Submit a pull request PR<br/><br/>- Documentation included<br/>- Tests implemented<br/>- CI checks passed]
+        B[Author: Propose a reviewer<br/><br/>- Default assignee: an OST rep for a quick review]
+        
+        A-->B
+        B-->C[Reviewer: Receive code review request]
+        C --> |No assistance needed| E[Reviewer: Execute code review]
+        C --> |Help wanted| D[Reviewer: Add an additional reviewer to the PR] 
+        D --> E
+        E --> |Changes needed| F[Reviewer: Request changes] 
+        F --> G[Author: Make changes]
+        G --> H[Reviewer: Approve PR]
+        E --> |No changes needed| H
+        H --> I[Author: Rebase and merge PR]
+ 
+        style A text-align:left, fill:#FFF
+        style B fill:#FFF
+        style G fill:#FFF
+        style I fill:#FFF
+", height = 1000)
 
-### Issue Templates
+```
 
-Templates are available and stored within each repository to guide users through the process of submitting a new issue.
-Example templates for issues can be found on
-[GitHub Docs](http://www.docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests).
-Use these references and existing templates stored in .github/ISSUE_TEMPLATE for reference when creating a new template.
+##### Assigning Reviewers
 
-## Branch Workflow
+Reviewers of PRs for changes to the codebase in FIMS should be suggested by the author of the PR. For those FIMS Implementation Team Members that keep their status in GitHub current (see ["Setting a status"](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/customizing-your-profile/personalizing-your-profile#setting-a-status) for more information), authors of PRs can use the status information to prevent assigning a reviewer who is known to be "Busy".
 
-This section details the workflow to create a branch in order to contribute to FIMS.
+If a review has been assigned to you and you don't feel like you have the expertise to address it properly, please respond directly to the PR so a different reviewer can be found promptly or suggest someone to help you with the review.
 
-### Branching Good Practices
+##### Review Checklist
 
-The following suggestions will help ensure optimal performance of the trunk-based branching strategy:
+While automated testing can assure the code structure and logic pass quality checks, human reviewers are required to evaluate things like functionality, readability, etc. Every PR is accompanied by an automatically generated checklist of major considerations for code reviews. Additional guidance is provided below for reviewers to evaluate when providing feedback on code:
```

Comment:
I wonder if we need to provide additional guidance in the book, Maybe we can just provide a link to the [Checklist in the FIMS](https://github.com/NOAA-FIMS/FIMS/blob/main/.github/workflows/pr-checklist.yml#L31C12-L41C74) repo to avoid updating the list in two places.

Reply:
This is a great idea @Bai-Li-NOAA. Though, I think that I will just link to the .yml file itself rather than specific lines because the whole file is relevant and I would hate to see the link to specific lines break if we were to edit that file. Does that sound okay?

Reply:
@Bai-Li-NOAA how much of this section should I delete? Should all bulleted points below line 178 be deleted and the last sentence of line 178?

Reply:
Yes, that's exactly what I had in mind. Removing all bulleted points below line 178 + last sentence of line 178.